### PR TITLE
fix: return kerrors' ErrNoMoreInstance when do discovery but no instance

### PR DESCRIPTION
**Pull request by codingxh**
- en: 
   -  Replace docker image to solve the problem that the service discovery after service registration in unit testing and the instance list is delayed by 30s
   -  Remove `time.Sleep(30 * time.Second)` in unit tests
   -  Correct argument position in `assert.Equal()` assertion
   -  Modify the `error` that needs to be returned when the instance is empty to `kerrors.ErrNoMoreInstance`
- zh: 
   -  替换docker镜像去解决单元测试中服务注册后进行服务发现拉取实例列表延迟30s的问题
   -  去除单元测试中的`time.Sleep(30 * time.Second)`
   -  纠正`assert.Equal()`断言中的参数位置
   -  修改当实例为空时所需要返回的`error`为`kerrors.ErrNoMoreInstance`

### CODE REVIEW

**Comment by codingxh** [comment]
hello,everyOne,this mr need the first mr (https://github.com/kitex-contrib/registry-eureka/pull/7) succeed to approve ci

File: resolver/resolver.go [hunk 1]
```diff
@@ -48,20 +49,18 @@ func (r *eurekaResolver) Target(ctx context.Context, target rpcinfo.EndpointInfo
 func (r *eurekaResolver) Resolve(ctx context.Context, desc string) (discovery.Result, error) {
 	application, err := r.eurekaConn.GetApp(desc)
 	if err != nil {
+		if errors.As(err, &fargo.AppNotFoundError{}) {
```
Comment (li-jin-gou) [inline resolver/resolver.go:52]: 这里为什么不用 errors.Is ??

File: docker-compose.yml [hunk 2]
```diff
@@ -17,6 +17,6 @@ version: '3'
 
 services:
   eureka:
```
Comment (li-jin-gou) [inline docker-compose.yml:19]: 空行加一下吧

File: discovery_test.go [hunk 3]
```diff
@@ -16,11 +16,12 @@ package test
 
 import (
 	"context"
-	"io"
 	"net"
 	"testing"
 	"time"
 
+	"github.com/cloudwego/kitex/pkg/kerrors"
```
Comment (li-jin-gou) [inline discovery_test.go:23]: 麻烦,更新一下 kitex 的版本吧,感谢。

File: docker-compose.yml (hunk 2 above)
Comment (codingxh) [reply docker-compose.yml:19]: no problem

**Review by simon0-o — CHANGES_REQUESTED** [review]

File: discovery_test.go [hunk 4]
```diff
@@ -16,11 +16,12 @@ package test
...
+
```
[the remark this answers is not in the document]
Comment (codingxh) [reply discovery_test.go:24]: OK

[the remark this answers is not in the document]
Comment (codingxh) [reply discovery_test.go]: hello, @simon0-o, it's done

**Review by GuangmingLuo — APPROVED** [review]

**Review by li-jin-gou — APPROVED** [review]

File: discovery_test.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (li-jin-gou) [reply discovery_test.go]: @codingxh  24 的空行去掉了,但是下面还有
预期效果应该是这样:
```
	"context"
	"io"
	"net"
	"testing"
	"time"

	"github.com/cloudwego/kitex/pkg/registry"
	"github.com/cloudwego/kitex/pkg/rpcinfo"
	registry2 "github.com/kitex-contrib/registry-eureka/registry"
	"github.com/kitex-contrib/registry-eureka/resolver"
	"github.com/stretchr/testify/assert"
```

**Pull request merged**